src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/OnboardingSupervisor.kt

```diff
@@ -73,21 +83,57 @@ internal class OnboardingSupervisor(
 
     private fun retrieveAssetsFromRouter() {
         if (StatsigConfig.useSkip) {
-            retrieveSkipTransferChains()
+            MainScope().launch {
```

Comment:
scopes is where things get tricky. a coroutine scope does not only define the lifecycle within which a coroutine will be run, but also a ton of other metadata such as which CoroutineDispatcher will be used to run it. `MainScope` runs on `Dispatchers.Main` (which runs coroutines on the main thread), but since we're updating Abacus state, we ideally want to be on the Abacus thread as defined by the client. 

I think `Dispatchers.Unconfined` is what we want here, since it'll execute the coroutine in the current thread. But generally, Unconfined is considered a bit of a code smell...